core/main/src/firebolt/handlers/provider_registrar.rs

```diff
@@ -296,73 +296,69 @@ impl ProviderRegistrar {
             let mut params_sequence = params.sequence();
             let call_context: Option<CallContext> = params_sequence.next().ok();
 
-            let mut event_data: Value = match params_sequence.next() {
+            let event_data: Value = match params_sequence.next() {
                 Ok(r) => r,
                 Err(e) => {
                     error!("callback_app_event_emitter: Error: {:?}", e);
                     return Err(Error::Custom("Missing event_data".to_string()));
                 }
             };
 
```

Comment:
Unrelated but cleaned-up this block a bit to also support events that contain a "non-object" payload, if we ever need it.